### PR TITLE
[Google Blockly] Remove padding in blocks rendered to read only workspaces in Dance

### PR DESCRIPTION
**Pull request by mikeharv**
Blocks in instructions cut off on https://studio.code.org/s/dance-2019/lessons/1/levels/1?blocklyVersion=Google
During a Dance Party bug bash, it was discovered that in-line blocks in instructions were being cut off. This was due to a 16px `padding` value, typically used to for positioning blocks within the main workspace.
image

image
image

This space was last touched here: 
- https://github.com/code-dot-org/code-dot-org/pull/48137

The changes here allow `padding` to be 0 if there's no `mainBlockSpace`. This ensures that the block fits inside the SVGs as expected if they're in a read only blockspace. I also refactored some nasty nested ternary logic to be hopefully more readable and added some comments. (Every time I've had to touch this file I've started by getting really confused about what it was even doing!)

I checked for this to fix the reported in issue in several Dance levels using the Google Blockly flag and also looked for regressions. Cases I considered include:
* Blocks rendered in-line within text in instructions/hint
* Blocks on their own line within text in instructions/hint
* Standard blocks on the workspace.
* Blocks with SVG frames (ie. function definitions) on the workspace.
* All of the cases as they appear in RTL mode

### LTR

http://localhost-studio.code.org:3000/s/dance-2019/lessons/1/levels/1?blocklyVersion=Google
image
http://localhost-studio.code.org:3000/s/dance-2019/lessons/1/levels/4?blocklyVersion=Google
image
http://localhost-studio.code.org:3000/flappy/1
image
http://localhost-studio.code.org:3000/s/poem-art-2021/lessons/1/levels/1
image
image

### RTL

http://localhost-studio.code.org:3000/s/dance-2019/lessons/1/levels/1?blocklyVersion=Google&lang=ar-SA
image
http://localhost-studio.code.org:3000/s/dance-2019/lessons/1/levels/4?blocklyVersion=Google&lang=ar-SA
image
http://localhost-studio.code.org:3000/flappy/1?lang=ar-SA
image
http://localhost-studio.code.org:3000/s/poem-art-2021/lessons/1/levels/1?lang=ar-SA
image
image

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by breville** [comment]
Likely out of scope, but it feels a bit odd that there is so much more room above the block than below it in this screenshot:

Screenshot 2022-11-09 at 1 16 21 PM

File: apps/src/blockly/addons/cdoXml.js [hunk 1]
```diff
@@ -86,14 +91,19 @@ export default function initializeBlocklyXml(blocklyWrapper) {
...
       const frameSvgMargin = hasFrameSvg ? 16 : 0;
 
+      // If the block doesn't already have coordinates, use the cursor.
       if (isNaN(block.x)) {
         block.x = cursor.x;
+      } else if (blockSpace.RTL) {
+        // Position RTLs with coordinates from the left.
+        block.x = viewWidth - block.x;
+      }
+      // Adjust for Svg Frames for function defintion blocks
```
Comment (breville) [inline apps/src/blockly/addons/cdoXml.js]: super nit: missing `.` :)

Comment (breville) [reply apps/src/blockly/addons/cdoXml.js]: also `defintion`

**Review by breville — APPROVED** [review]
It is confusing code but your comments do help, and the screenshots look like a really comprehensive set of coverage.

File: apps/src/blockly/addons/cdoXml.js [hunk 2]
```diff
@@ -67,17 +67,22 @@ export default function initializeBlocklyXml(blocklyWrapper) {
...
     let cursor = {
-      x: blockSpace.RTL
-        ? viewWidth
-          ? viewWidth - padding
-          : blockWidth + padding
-        : padding,
+      x: padding,
       y: padding
     };
+    if (blockSpace.RTL && viewWidth) {
```
Comment (fisher-alice) [inline apps/src/blockly/addons/cdoXml.js:79]: I appreciate how you replaced nested ternary statements here.

File: apps/src/blockly/addons/cdoXml.js [hunk 3]
```diff
@@ -86,14 +91,19 @@ export default function initializeBlocklyXml(blocklyWrapper) {
       const frameSvgTop = hasFrameSvg ? 35 : 0;
```
Comment (fisher-alice) [inline apps/src/blockly/addons/cdoXml.js:91]: Not part of this PR, but there are a lot of numbers here that seem mysterious - just wondering how they're decided upon.

**Review by fisher-alice — APPROVED** [review]
Nice job and great coverage of testing cases. Personally, I go back and forth between wanting to use terneries or not. Overall, your code was clear to me.

File: apps/src/blockly/addons/cdoXml.js (hunk 3 above)
Comment (mikeharv) [reply apps/src/blockly/addons/cdoXml.js:91]: Agreed.
